src/openzeppelin/tests/token/test_erc721.cairo

```diff
@@ -1342,3 +1423,26 @@ fn assert_state_after_mint(token_id: u256, recipient: ContractAddress) {
     assert(ERC721Impl::balance_of(@state, recipient) == 1, 'Balance of recipient after');
     assert(ERC721Impl::get_approved(@state, token_id) == ZERO(), 'Approval implicitly set');
 }
+
+fn assert_event_approval_for_all(
+    owner: ContractAddress, operator: ContractAddress, approved: bool
+) {
+    let event = testing::pop_log::<ApprovalForAll>(ZERO()).unwrap();
+    assert(event.owner == owner, 'Invalid owner');
+    assert(event.operator == operator, 'Invalid operator');
+    assert(event.approved == approved, 'Invalid approved');
+}
+
+fn assert_event_approval(owner: ContractAddress, approved: ContractAddress, token_id: u256) {
+    let event = testing::pop_log::<Approval>(ZERO()).unwrap();
+    assert(event.owner == owner, 'Invalid owner');
+    assert(event.approved == approved, 'Invalid approved');
+    assert(event.token_id == token_id, 'Invalid token_id');
+}
+
+fn assert_event_transfer(from: ContractAddress, to: ContractAddress, token_id: u256) {
+    let event = testing::pop_log::<Transfer>(ZERO()).unwrap();
+    assert(event.from == from, 'Invalid from');
+    assert(event.to == to, 'Invalid to');
+    assert(event.token_id == token_id, 'Invalid token_id');
+}
```

Comment:
are these really asserts? they look more like checks or validations, no? anyway no strong opinions

Reply:
To me assert looks ok because is panicking if the event is incorrect (is making assertions over the event fields).